.github/workflows/go-test.yml

```diff
@@ -0,0 +1,28 @@
+name: go test
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  build:
+    name: go test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.20"
+      - uses: actions/cache@v3
+        with:
```

Comment:
I'm curious what motived this with paragraph. I didn't see it in the official doc: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go

Reply:
Idea is to cache the go mod folders for faster incremental builds. Happy to omit it

Reply:
Docs here: https://github.com/actions/cache/blob/main/examples.md#linux-1

Reply:
wonderful. can you add that link as a comment in the file?

Reply:
done!